### ISSUE TITLE
Reordering notes in custom sort doesn't work

### ISSUE THREAD

**Issue by annoyedhumanoid** [laurent22/joplin#11226]
### Operating system

macOS

### Joplin version

3.0.15

### Desktop version info

Joplin 3.0.15 (prod, darwin)

Client ID: 11cb0a2d420047cd9119a44d443e827c
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: 598677b

Backup: 1.4.1

### Current behaviour

1. Create a notebook with several notes
2. Toggle sort order field to custom
3. Drag a note up or down in the note list sidebar to move it
4. It jumps to the top and can only be moved between other notes that have jumped to the top

https://github.com/user-attachments/assets/978a3d9d-d18e-4c22-88e7-3da75ef28eac

### Expected behaviour

The dragged notes should move to the position hinted by the horizontal line between notes, rather than sticking to the top.

### Logs

The only message in the console associated with moving a note is the following:
```
App: node:internal/process/task_queues:95
Refreshing notes: 2 5ae3997be9a94aeab4f21e069fd97d01
```

**Comment by personalizedrefrigerator** [comment]
I can reproduce this on Linux with v3.1.18 and v3.0.15. For me, alt+arrow keys seem to correctly re-order notebooks, but clicking and dragging fails.

**Comment by Nirvanjha2004** [comment]
https://github.com/user-attachments/assets/ed91dadc-b73f-4e9f-a4df-b2666e072ec2

In Windows its working fine

**Comment by CaptainPackers** [comment]
This doesn't work on my Android 12 Motorola moto g stylus 5G or my Android 14 Samsung Galaxy Tab S6 Lite.

**Comment by personalizedrefrigerator** [comment]
I'm documenting how the note sort `order` property changes when I experience a similar issue locally.

When I try to drag a note "A" to the next position in the list,
- "A" is initially near the end of the note list with `order=0.0002919084705921691`.
- Notes with smaller `order` are placed closer to the end of the note list.
- After dragging, its order decreases to `order=0.00029190265035641594`.
- The next note in the list has `order=0.0002918968301206628`, which is still **less** than note A's new `order` field.
  $$\texttt{noteA.order} - \texttt{nextNote.order} = 0.00029190265035641594 - 0.0002918968301206628\\ \approx 5.820235753149338 \cdot 10^{-9}$$

Based on this, I suspect that the issue is in [`Note.insertNotesAt`](https://github.com/laurent22/joplin/blob/3cba4ec82cc38b8314e37e46f5fd82984054fe2c/packages/lib/models/Note.ts#L1010).

**Comment by personalizedrefrigerator** [comment]
**Note**: [As reported here](https://discourse.joplinapp.org/t/deleting-pages-breaks-page-reordering/42856), it may be necessary to first move one or more items to the trash before it's possible to reproduce the issue. The reproduction steps might thus be:
1. Create a notebook with several notes.
2. Switch to a custom sort order.
3. Move several notes to the trash folder.
4. Drag and drop a note in the note list sidebar near the location of the deleted note.

**Comment by lukewlms** [comment]
@personalizedrefrigerator yes - I think that explains why some people repro and some don't.  In a totally fresh folder it never repros; the bug appears once some notes have been moved to trash.

Thanks for linking.  The linked issue contains full repro steps which I suspect anyone can use to consistently repro the issue.

(This issue has been driving me crazy so I'd been trying on and off to find a consistent repro!)

**Comment by lukewlms** [comment]
My current workaround is working beautifully:
1. Use Keyboard Maestro macro
2. On Delete, focus the note list and hit cmd+delete

Then in the background, via Joplin API:
3. Fetch the last 500 (sanity max) most recently updated Notes (including deleted)
4. For any that are not in my "Intermediate Trash" folder, move them there

This ensures that any items in the trash always have parent_id pointing to Intermediate Trash.  This has a small disadvantage that when I restore them they would restore to that folder (which always appears empty otherwise), but it allows me to instantly delete pages while also avoiding this reordering bug.

Just to verify that this approach seems highly reliable so far and further confirms the likely root of the issue (that Trashed pages are breaking reordering operations for their parent folder).
